sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -5016,6 +5016,10 @@ piEnqueueKernelLaunch(pi_queue Queue, pi_kernel Kernel, pi_uint32 WorkDim,
   PI_ASSERT(Event, PI_ERROR_INVALID_EVENT);
   PI_ASSERT((WorkDim > 0) && (WorkDim < 4), PI_ERROR_INVALID_WORK_DIMENSION);
 
+  // We need a command list to submit the command, that's why reset signalled
+  // command lists to avoid creating new one.
+  resetCommandLists(Queue);
```

Comment:
this way you are going to check all running fences at each enqueue request whereas previously this check/reset was only done when no command-list was available. should we add this check/shortcut to resetCommandLists?

Reply:
I agree, thanks for noticing. Unfortunately, one problem with just adding check/shortcut to resetCommandLists is that available command list may be occupied by another thread as soon as we unlock the queue mutex, so when we reach getAvailableCommandList for the current thread there will be no available command list again.

That's why I returned the old behavior back - reset command lists in the getAvailableCommandLists. It is ok in terms of mutex locks now because unnecessary locks were removed from retain/release functions in https://github.com/intel/llvm/commit/b545c60b38da9ba8246a740f88faf80f5bd48379